components/twitter/twitter.app.js

```diff
@@ -818,19 +885,212 @@ module.exports = {
+        q = `${q} -filter:nativeretweets`
+      }
+  
+      const response = await this.search(q, since_id, tweet_mode, count, result_type, lang, locale, geocode, max_id)
+      
+      //console.log(response)
+      if (_.get(response, 'status', 'Error') === 200) {
+        if (!max_id) {
+          max_id = since_id || 0
+        }
+    
+        for (let tweet of response.data.statuses) {
+          if (tweet.id_str !== since_id && tweet.id_str !== max_id) {
+            if (enrichTweets) {
+              tweet.created_at_timestamp = moment(tweet.created_at, 'ddd MMM DD HH:mm:ss Z YYYY').valueOf()
+              tweet.created_at_iso8601 = moment(tweet.created_at, 'ddd MMM DD HH:mm:ss Z YYYY').toISOString()
+            }
+            tweets.push(tweet)
+            if (tweet.id_str > max_id) {
+              max_id = tweet.id_str
+              if(!min_id) {
+                min_id = max_id
+              }
+            }
+            if (tweet.id_str < max_id) {
+              min_id = tweet.id_str
+            }
+          }
+        }
+
+        return {
+          tweets,
+          max_id,
+          min_id,
+          count,
+          resultCount: response.data.statuses.length,
+          statusCode: response.status, 
+        }
+      } else {
+        console.log(`Last request was not successful.`)
+        console.log(`API Response: ${response}`)
+        return {
+          statusCode: "Error",
+        }
+      }
+    },
+    async paginatedSearch(opts = {}) {
+      const {
+        count = 100,
+        q,
+        since_id,
+        lang, 
+        locale, 
+        geocode, 
+        result_type, 
+        enrichTweets,
+        includeReplies, 
+        includeRetweets,
+        maxRequests = 1,
+        limitFirstPage = true,
+      } = opts
+
+      let { max_id } = opts, maxPages = 1, totalRequests = 0
+
+      const tweets = []
+
+      if (!limitFirstPage) {
+        maxPages = maxRequests
+      }
+
+      //console.log(maxPages)
+
+      for (let page = 0; page < maxPages; page++) {        
+        //console.log(`page: ${page} max_id: ${max_id}`)
+        const response = await this.searchHelper({
+          count,
+          q,
+          since_id,
+          max_id,
+          lang, 
+          locale, 
+          geocode, 
+          result_type, 
+          enrichTweets,
+          includeReplies, 
+          includeRetweets,
+        })
+        
+        // increment the count of requests to report out after all requests are complete
+        totalRequests++
+        
+        if (response.statusCode !== 200) {
+          break
+        } else {
+          response.tweets.forEach(tweet => {
+            tweets.push(tweet)
+          })
+
+          //console.log(`total requests: ${totalRequests} max requests: ${maxRequests} results: ${response.resultCount} count: ${response.count}`)
+
+          if (totalRequests * 1 === maxRequests * 1 && response.resultCount === response.count) {
```

Comment:
what in the world is this advanced mathematics

Reply:
i changed this to parseInt() -- lmk if there's a better option